build/Dockerfile

```diff
@@ -4,170 +4,301 @@
+
+# =============================================================================
+# TARGET: rollups-node-ci
+#
+# This stage prepares the base CI image
 #
-# This target contains the Cartesi machine snapshot used in the rollups-node integration and
-# end-to-end tests.
+# - Install git and build-essential (setup).
+# - Set CGO_ENABLED.
+# - Install docker.
+# =============================================================================
+
+FROM emulator-devel as rollups-node-ci
+
+# Install git and build-essential (setup).
+RUN <<EOF
+    set -e
+    apt-get update
+    apt-get install -y --no-install-recommends git-all build-essential
+EOF
 
+ENV CGO_ENABLED=1
+
+# Install docker.
+RUN <<EOF
+    install -m 0755 -d /etc/apt/keyrings
+    URL=https://download.docker.com/linux/debian
+    curl -fsSL ${URL}/gpg -o /etc/apt/keyrings/docker.asc
+    chmod a+r /etc/apt/keyrings/docker.asc
+    ARCH=$(dpkg --print-architecture)
+    SIGNED_BY=/etc/apt/keyrings/docker.asc
+    echo "deb [arch=${ARCH} signed-by=${SIGNED_BY}] ${URL} $(. /etc/os-release && echo "$VERSION_CODENAME") stable" \
+        | tee /etc/apt/sources.list.d/docker.list > /dev/null
+    apt-get update
+    apt-get install -y --no-install-recommends \
+        docker-ce \
+        docker-ce-cli \
+        containerd.io \
+        docker-buildx-plugin \
+        docker-compose-plugin
+    # Cartesi Machine Emulator
+    SDK_URL=https://github.com/cartesi/machine-emulator-sdk
+EOF
+
+# =============================================================================
 # STAGE: snapshot-builder
 #
-# This stage builds the snapshot using the machine emulator as base image.
-FROM emulator-base as snapshot-builder
-
-# Download rootfs and linux.
-# Add these files to the directories the cartesi-machine expects.
-WORKDIR /usr/share/cartesi-machine/images/
-ARG TOOLS_VERSION
-ARG LINUX_VERSION
-ARG LINUX_KERNEL_VERSION
-ADD https://github.com/cartesi/machine-emulator-tools/releases/download/v${TOOLS_VERSION}/rootfs-tools-v${TOOLS_VERSION}.ext2 rootfs.ext2
-ADD https://github.com/cartesi/image-kernel/releases/download/v${LINUX_VERSION}/linux-${LINUX_KERNEL_VERSION}.bin linux.bin
+# This stage builds an echo snapshot.
+# =============================================================================
+
+FROM emulator as snapshot-builder
 
 # Generate snapshot with echo and store it.
 WORKDIR /build
 ARG SNAPSHOT_BUILD_PATH
 RUN cartesi-machine \
     --ram-length=128Mi \
     --store=$SNAPSHOT_BUILD_PATH \
+    --no-rollback \
     -- "ioctl-echo-loop --vouchers=1 --notices=1 --reports=1 --verbose=1"
 
-# STAGE: rollups-node-snapshot
+# =============================================================================
+# TARGET: rollups-node-snapshot
 #
-# This stage copies the image from the builder.
-# We use the emulator as base image so we can easily create a container with a volume shared with
-# the rollups-node container.
-FROM emulator-base as rollups-node-snapshot
+# TODO: this should be deleted.
```

Comment:
Please remove the TODO. What's the reason behind it? We should either cover this in a separate ticket.

Reply:
Serves as a reminder (and temporary documentation), that we should not create snapshots in the CI anymore. I'm going to update it to add this information.